File: CustomerApp/app/src/main/java/com/example/customerapp/data/models/Customer.kt
package com.example.customerapp.data.models

import com.google.gson.annotations.SerializedName


data class Customer(
    @SerializedName("first_name") val firstName: String,
    @SerializedName("last_name") val lastName: String,
    @SerializedName("phone") val phone: String,
    @SerializedName("pin") val pin: String,
    @SerializedName("gender") val gender: String,
    @SerializedName("email") val email: String,
    @SerializedName("county") val county: String,
    @SerializedName("sub_county") val subCounty: String
)
data class LoginRequest(
    val phone: String,
    val pin: String
)

data class LoginResponse(
    val token: String,
    @SerializedName("customer") val customer: CustomerWithId
)

data class CustomerWithId(
    @SerializedName("id") val id: Int,
    @SerializedName("first_name") val firstName: String,
    @SerializedName("last_name") val lastName: String,
    @SerializedName("phone") val phone: String,
    @SerializedName("pin") val pin: String,
    @SerializedName("gender") val gender: String,
    @SerializedName("email") val email: String,
    @SerializedName("county") val county: String,
    @SerializedName("sub_county") val subCounty: String,
    @SerializedName("created_at") val createdAt: String,
    @SerializedName("updated_at") val updatedAt: String,
    @SerializedName("status") val status: String
)

File: ride-hailing/app/Models/Customer.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

use Illuminate\Foundation\Auth\User as Authenticatable;

use Laravel\Sanctum\HasApiTokens;


class Customer extends Authenticatable
{
    use HasApiTokens,HasFactory;

    protected $fillable = [
        'id',
        'first_name',
        'last_name',
        'phone',
        'pin', // Assuming it's hashed
        'gender',
        'email',
        'county',
        'sub_county',
    ];

    // Relationship with Ride
    public function rides()
    {
        return $this->hasMany(Ride::class);
    }
}
